packages/solana/src/client.ts

```diff
@@ -227,6 +229,17 @@ export class Web3Modal extends Web3ModalScaffold {
       }
     })
 
+    EventsController.subscribe(state => {
+      if (state.data.event === 'SELECT_WALLET' && state.data.properties?.name === 'Phantom') {
+        const isMobile = CoreHelperUtil.isMobile()
+        if (isMobile && !window.phantom) {
```

Comment:
In server-side frameworks like Next.js, reaching to `window` object will possibly throw error since it's not available on the server side. So we should make an extra check for the window object before using it to make sure this condition is running only on the client side. @phoenixVS